src/main/java/io/jenkins/pluginhealth/scoring/http/ProbeResultController.java

```diff
@@ -0,0 +1,54 @@
+package io.jenkins.pluginhealth.scoring.http;
+
+import javax.transaction.Transactional;
+import java.util.HashMap;
+import java.util.Map;
+
+import io.jenkins.pluginhealth.scoring.model.ResultStatus;
+import io.jenkins.pluginhealth.scoring.service.PluginService;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.servlet.ModelAndView;
+
+@Controller
+@RequestMapping(path = "/results")
+public class ProbeResultController {
+
+    private final PluginService pluginService;
+
+    public ProbeResultController(PluginService pluginService) {
+        this.pluginService = pluginService;
+    }
+
+    @GetMapping(path = "")
+    public ModelAndView list() {
+        final ModelAndView modelAndView = new ModelAndView("probes/results");
+
+        modelAndView.addObject(
+            "probeResults",
+            getProbeResultsData()
+        );
+
+        return modelAndView;
+    }
+
+    @Transactional
+    public Map<String, Integer> getProbeResultsData() {
+        Map<String, Integer> probeResultsMap = new HashMap<>();
+
+        pluginService.streamAll().forEach(
```

Comment:
To collect the probe results data, I'm relying on the plugins stored in the database by performing an additional query to the DB. I scan their `details` value and accordingly populate the `probeResultsMap`. Is this an acceptable approach?

An alternate approach could be to populate this map while we're running the probe engine, in this way we wouldn't need to query the DB twice (as done in the above-mentioned approach). But later I realized that this might not be an ideal approach because when we're not running our probe engine, the `probeResultsMap` map would be empty and so will our `/results` page. Plus, when we're running the probe engine, this map will get populated, but this will very happen very slowly as the probe engine takes some time in processing each probe.

wdyt?

Reply:
> To collect the probe results data, I'm relying on the plugins stored in the database by performing an additional query to the DB.

I don't understand this. Could you please detail what you mean?

Reply:
> I don't understand this. Could you please detail what you mean?

Yes, sorry I should've mentioned that I was referring to the `pluginService.streamAll()` (used in line 41 above) method which does an "additional query" to the DB. So I thought about maybe there's a way where we'd not even need to query the DB, and then I started thinking about the "alternate approach" as I mentioned above. But I can see now why this alternate approach is a bad idea.


> Which means we wouldn't need to have the DB and we would have everything in memory. I don't think this is a good idea.
> 

:smile: true. I need to keep in mind that having everything in memory is not good, and there's a reason we have concept of DB.


> To get the number to Jake, I ran this SQL query:
> 
> ```
> SELECT
>     count(id) as total,
>     (SELECT count(id) FROM plugins WHERE scm IS NOT NULL AND details -> 'up-for-adoption' ->> 'status' = 'FAILURE') as upForAdoption,
>     (SELECT count(id) FROM plugins WHERE scm IS NOT NULL AND details -> 'security' ->> 'status' = 'FAILURE') as security,
>     (SELECT count(id) FROM plugins WHERE scm IS NOT NULL AND details -> 'deprecation' ->> 'status' = 'FAILURE') as deprecated,
>     (SELECT count(id) FROM plugins WHERE scm IS NOT NULL AND details -> 'dependabot' ->> 'status' = 'SUCCESS') as dependabot,
>     (SELECT count(id) FROM plugins WHERE scm IS NOT NULL AND details -> 'jep-229' ->> 'status' = 'SUCCESS' ) as jep229,
>     (SELECT count(id) FROM plugins WHERE scm IS NOT NULL AND details -> 'jenkinsfile' ->> 'status' = 'SUCCESS') as jenkinsfile
> FROM plugins
> WHERE scm IS NOT NULL;
> ```
> 
> Maybe you can map that to a (or multiple) method signature in `PluginRepository` and have a method in `PluginService` to return that map.

Wow, this is the first time I saw an SQL query being run on a JSON object. Thanks for sharing the SQL code. I'll think about this now.